### PR TITLE
Refactor login form

### PR DESCRIPTION
**Pull request by lednhatkhanh**
Please take a look:

- Note: Small but not too small, I really like the way you organize your project, but don't split the component into too small chunks in a single file and pass props everywhere, components should be able to be re-used. Doing the way you are doing will actually make the code harder to debug when there are errors happen with nested props!
- Note:  I may removed/added some unwanted/vital styles, please take a look while merging this PR

What do we have:
- Easier to validate the forms with yup
- Less dependent on redux (redux-form will trigger the WHOLE redux tree on EVERY key stroke and it's really bad for react-native app which needs a lot of performance optimizations)
- Yup and formik are way more smaller and faster than redux-form and redux
- A cleaner redux-devtools :)
- No need to build custom Input components anymore
- No need state to hold the form state anymore.

@littlecrow Please take a look and do the same for other forms, after that remove the redux-form completely 💯

### CODE REVIEW

**Comment by pvtri96** [comment]
Yup is really cool! Validation is much cleaner. 
However, about this line
```
No need to build custom Input components anymore
```
How about the component that is NOT an Input. 
For example: we are currently having a component named DatePicker, you can easily find it in `~/Component`.

**Comment by lednhatkhanh** [comment]
Stay the same with just a little bit of hacking just like what I doing right now with the input, set the value of it by these functions
```
setFieldError: (field: string, errorMsg: string) => void
setFieldTouched: (field: string, isTouched: boolean) => void
setFieldValue: (field: string, value: any) => void
```
And you should be good to go

File: App/Scene/Auth/Login/Form.js [hunk 1]
```diff
@@ -26,101 +33,141 @@ const _renderHeaderImage = () => (
...
-      {_renderError(loginError)}
-      {_renderButton({ handleSubmit, onLogin, loading })}
-      {_renderFooter(onNavigate)}
+      <View style={styles.formContainer}>
+        <FormLabel>Email</FormLabel>
+        <FormInput
+          name="email"
+          value={values.email}
+          onChangeText={text => {
+            setFieldValue('email', text);
```
Comment (pvtri96) [inline App/Scene/Auth/Login/Form.js]: I can see you are repeating these line many times. How can you figure out when a form that is NOT simple to this Login form (e.g. User Profile have more than 10 input). We need to make the code look cleaner.

**Comment by lednhatkhanh** [comment]
Make a custom function which received a name and value the call the functions for that :)
This is just a initial refactor state for the component, you should think about some workarounds/refactor for bigger forms

**Comment by pvtri96** [comment]
Can you build a Component that receives only 1 prop (or maybe more) that is the name of the Input? It will be good to learn some code patterns from you.

**Comment by lednhatkhanh** [comment]
Let's keep this PR around, I'll make more modifications tomorrow.

**Pull request closed without merging**